### ISSUE TITLE
Error in kmers_add_strand_information: empty add_strand.log.out file

### ISSUE THREAD

**Issue by douwetb** [voichek/kmersGWAS#77]
Dear Y. Voicheck,

Thank you for the interesting tool. I've been using your tool and it worked fine for all steps including the final kmers_gwas.py script. However, I only just realized my add_strand.log.out files are empty, while I was trying to gather usefull statistics about the KMC run. I've think I'm having a similar problem as a previous issue #40 on the github (https://github.com/voichek/kmersGWAS/issues/40). 

I re-run the command for only one sample to see what my problem could be. My command got killed after 5-10 minutes of running: /bin/kmers_add_strand_information -c $kmc_canon -n $kmc_non_canon -k 31 -o $output_dir/kmers_with_strand > $output_dir/add_strand.log.out"

First I looked whether the problem occured at previous command lines. Here I discovered that my kmc_canon.log.out file looks fine, but my kmc_non_canon.log.out shows missing k-mers. I was wondering whether you know what caused this problem?
I haven't included the raw kmc files due to their sizes.

[kmc_non_canon_log_out.txt](https://github.com/voichek/kmersGWAS/files/5887415/kmc_non_canon_log_out.txt)
[kmc_canon_log_out.txt](https://github.com/voichek/kmersGWAS/files/5887416/kmc_canon_log_out.txt)

Furthermore I have a different question. I can also create a new issue if you think this would be more convenient documentation-wise.   
I would like to obtain the textual presence/absence table for all k-mers that were created during the KMC run. I'm capable of doing this with the **filters_kmers**, but I lack a list of all k-mers that were counted. Do you know whether there is a option to retrieve all these k-mers? So far I only have a list of the k-mers that passed the threshold. 

Kind regards,
Douwe

**Comment by voichek** [comment]
Dear Douwe,

**_Regarding the first question_** - What do you mean missing k-mers ? I see the total number of k-mers is the same in both files (32866860109). 
- Can you copy to here the commands you used to run KMC?
- Did the "kmers_add_strand_information" give any error message?
- Ideally it would be best if you can upload to somewhere the two KMC files so I can try to recreate the problem on my system.
- Did you try to run the second example (with e. coli)? - to see if you also have a problem there.

_**Regarding the second question**_ - I am not sure I fully understand this question. You want the presence absence of all k-mers and not only the ones which are part of the _k-mers_ table? If so you can just lower the threshold (put the threshold to zero) for a parallel creation of the _k-mers_ table for this purpose. If you want the the full list of _k-mers_ in the _k-mers_ table you can use the conversion of the _k-mers_ table to plink binary format and you will have it in the bim file.

Best,
Yoav

**Comment by douwetb** [comment]
Dear Yoav, thank you for the reply. 

First of all, I noticed that the kmers_add_Strand file and the add_strand.log.out were not (properly) created due to a lack in RAM memory. This is now fixed.

I was indeed looking for a way to get all k-mers that are in the k-mer table. But of course this can be retrieved by converting it to PLINK format and then extract the sequences from the .bim files. Thus you've helped perfectly again, thank you.
